lib/index.js

```diff
@@ -415,17 +423,40 @@ async function run(options = {}) {
   }
 
   async function getAnalysis() {
+    const defaultPackageFilename = getPackageFileName(options)
+    const pkgs = globby.sync(defaultPackageFilename)
+    options.deep = options.deep || pkgs.length > 1
+
+    let analysis
     if (options.global) {
-      const analysis = await analyzeGlobalPackages(options)
-      clearTimeout(timeout)
-      return analysis
+      analysis = await analyzeGlobalPackages(options)
+    }
+    else if (options.deep) {
+      const deepJsonOut = {}
+      analysis = await pkgs.reduce(async (previousPromise, packageFile) => {
+        const packages = await previousPromise
+        if (packageFile.length > 0 && packageFile !== defaultPackageFilename) {
+          options.packageFile = packageFile // set packageFile only if needed
+        }
+        const [pkgData, pkgFile] = await findPackage(options)
+        return {
+          ...packages,
+          [pkgFile]: await analyzeProjectDependencies(options, pkgData, pkgFile, deepJsonOut)
```

Comment:
Since the `reduce` returns the aggregate output, we should be able to pass that directly to `printJson` rather than using `deepJsonOut`, unless I'm missing something. Either way, I consider passing a mutable object to a subcall to modify to be a bad practice. Let's try to find an immutable solution.

Reply:
Only reason currently is to display well formatted `json` output.

```
{
  "package.json": {
    ...
  },
  "pkgs/pkg1/package.json": {
    ...
  }
  ...
}

```


https://github.com/raineorshine/npm-check-updates/blob/59cae824de738767e3ffcba02151be5f9825bb25/lib/logging.js#L38-L42

Reply:
Yes, that is the intended functionality. I'm not speaking of the functionality, but rather the implementation. I consider mutations a bad practice. Unfortunately passing a value by reference and mutating it is unacceptable for the coding standards of this project. I understand and respect that approach for other situations, but just not here. You'll have to find a way to achieve the same functionality but follow a more functional approach that respects the value of immutability.

Reply:
my bad missing that solution to follow that practice was super simple